relay/pkg/node/node.go

```diff
@@ -1,20 +1,56 @@
 package node
 
 import (
+	"context"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/tbtc/relay/pkg/block"
 	"github.com/keep-network/tbtc/relay/pkg/btc"
 	"github.com/keep-network/tbtc/relay/pkg/chain"
-	"github.com/keep-network/tbtc/relay/pkg/forwarder"
 )
 
+var logger = log.Logger("relay-node")
+
 // Initialize initializes the relay node.
+//
+// TODO: This function will be probably the right place to handle new
+//  requirements which will require starting and stopping the block forwarder.
 func Initialize(
+	ctx context.Context,
 	btcChain btc.Handle,
 	hostChain chain.Handle,
-) error {
-	err := forwarder.Initialize(btcChain, hostChain)
-	if err != nil {
-		return err
-	}
+) {
+	logger.Infof("initializing relay node")
 
-	return nil
+	go runForwarderControlLoop(ctx, btcChain, hostChain)
+}
+
+// runForwarderControlLoop runs a block forwarder control loop which is
+// responsible for starting the forwarder and acting upon errors by restarting
+// the forwarder instance. The lifecycle of the control loop itself can
+// be managed using the passed context.
+func runForwarderControlLoop(
+	ctx context.Context,
+	btcChain btc.Handle,
+	hostChain chain.Handle,
+) {
+	logger.Infof("running forwarder control loop")
+
+	for {
+		logger.Infof("running new forwarder instance")
+
+		forwarder := block.RunForwarder(ctx, btcChain, hostChain)
+
+		select {
+		case err := <-forwarder.ErrChan():
+			logger.Errorf(
+				"forwarder terminated with error: [%v] "+
+					"and will be restarted immediately",
+				err,
+			)
```

Comment:
Maybe just `"relay occurred an error: [%v]"`? It will be clear from the logs it is continuing its work. "Restart" feels a little too rough. As an operator, I'd expect the entire process is restarted/

Reply:
Agree however because of https://github.com/keep-network/tbtc/pull/767#discussion_r598995537 my proposition is: https://github.com/keep-network/tbtc/commit/c6fb5d71e9c976fb98ee676e36dfcd48ff6221fe

Reply:
As stated in https://github.com/keep-network/tbtc/pull/767#discussion_r603913978, the message will be focused around `headers` not `blocks`